Dockerfile

```
@@ -0,0 +1,13 @@
+FROM python:3.10.12 AS base
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH=$PATH:/root/.local/bin/
+COPY . /app
+WORKDIR /app
+RUN poetry install
```

This works nicely, but one option to consider for potential improvements is whether we can cache the `poetry install` step more often - that's probably the slowest layer to build, and it only really needs to change when we update our dependencies. One way we could do that is to split our COPY command:
```suggestion
COPY pyproject.toml poetry.lock /app
WORKDIR /app
RUN poetry install
COPY . /app
```
This way we only copy in the dependency-related files, and we copy everything else in afterwards